### PR TITLE
Refactor tests

### PR DESCRIPTION
**Pull request by kevinearls**
This PR is the first part of #385 and covers the smoke tests.  It includes restructuring tests and using the [stretchr/testify](https://github.com/stretchr/testify) toolkit to help produce better test reporting.

### CODE REVIEW

**Comment by kevinearls** [comment]
@objectiser @pavolloffay @jpkrohling  Please review

File: test/e2e/sidecar_test.go [hunk 1]
```diff
@@ -21,48 +24,80 @@ import (
...
-			Strategy: "allInOne",
-			AllInOne: v1.JaegerAllInOneSpec{},
-			Agent: v1.JaegerAgentSpec{
-				Options: v1.NewOptions(map[string]interface{}{
-					"log-level": "debug",
-				}),
-			},
-		},
-	}
+	j := getJaegerDefinition(namespace)
```
Comment (objectiser) [inline test/e2e/sidecar_test.go]: Should be named something like `getJaegerAgentAsSidecarDefinition` just to distinguish it from other methods also returning CRs.

File: test/e2e/sidecar_test.go [hunk 2]
```diff
@@ -21,48 +24,80 @@ import (
...
+	err := fw.Client.Create(goctx.TODO(), j, cleanupOptions)
+	require.Nil(t, err, "Failed to create jaeger-operator instance")
```
Comment (objectiser) [inline test/e2e/sidecar_test.go]: It isn't actually a jaeger-operator instance, but a jaeger instance.

File: test/e2e/all_in_one_test.go [hunk 3]
```diff
@@ -77,58 +74,41 @@ func allInOneWithIngressTest(t *testing.T, f *framework.Framework, ctx *framewor
 	}
 
 	log.Infof("passing %v", exampleJaeger)
-	err = f.Client.Create(goctx.TODO(), exampleJaeger, &framework.CleanupOptions{TestContext: ctx, Timeout: timeout, RetryInterval: retryInterval})
-	if err != nil {
-		return err
-	}
+	err := fw.Client.Create(goctx.TODO(), exampleJaeger, &framework.CleanupOptions{TestContext: ctx, Timeout: timeout, RetryInterval: retryInterval})
+	require.Nil(t, err, "Failed to create jaeger-operator instance")
```
Comment (objectiser) [inline test/e2e/all_in_one_test.go]: It should be `jaeger instance`, it is not the operator being created.

File: test/e2e/all_in_one_test.go [hunk 4]
```diff
@@ -77,58 +74,41 @@ func allInOneWithIngressTest(t *testing.T, f *framework.Framework, ctx *framewor
...
-		route, err := findRoute(t, f, name)
-		if err != nil {
-			return err
-		}
-
-		if len(route.Status.Ingress) != 1 {
-			return fmt.Errorf("Wrong number of ingresses. Expected 1, was %v", len(route.Status.Ingress))
-		}
+		route := findRoute(t, fw, name)
+		require.Equalf(t, route.Status.Ingress, 1, "Wrong number of ingresses. Expected 1, was %v", len(route.Status.Ingress))
```
[the remark this answers is not in the document]
Comment (kevinearls) [reply test/e2e/all_in_one_test.go]: I'm going to change this to use require.Len, which uses actual then expected.  https://github.com/stretchr/testify/blob/master/require/require.go#L795-L807

File: test/e2e/all_in_one_test.go [hunk 5]
```diff
@@ -26,47 +29,41 @@ import (
...
-	ctx := prepare(t)
-	defer ctx.Cleanup()
+type AllInOneTestSuite struct {
+	suite.Suite
+}
 
-	if err := allInOneTest(t, framework.Global, ctx); err != nil {
-		t.Fatal(err)
-	}
+func(suite *AllInOneTestSuite) SetupSuite() {
```
Comment (pavolloffay) [inline test/e2e/all_in_one_test.go]: Does the SDK clean up all resources (namespace..) when no returning an error?

Comment (kevinearls) [reply test/e2e/all_in_one_test.go]: @pavolloffay cleanup should happen in the TearDownSuite method where ctx.cleanup() is called.  This should work the same as previous tests did.

**Pull request closed without merging**